torchgeo/datasets/seco.py

```diff
@@ -88,57 +97,42 @@ def __init__(
             RuntimeError: if ``download=False`` and data is not found, or checksums
                 don't match
         """
-        assert version in ["100k", "1m"]
+        assert version in self.metadata.keys()
+        assert seasons in range(5)
         for band in bands:
             assert band in self.all_bands
 
         self.root = root
+        self.version = version
+        self.seasons = seasons
         self.bands = bands
-        self.url = self.urls[version]
-        self.filename = self.filenames[version]
-        self.md5 = self.md5s[version]
-        self.directory_name = self.directory_names[version]
         self.transforms = transforms
         self.download = download
         self.checksum = checksum
 
         self._verify()
 
-        # TODO: This is slow, I think this should be generated on download and then
-        # loaded in the constructor
-        self.scene_to_patches = defaultdict(list)
-        for root_directory, directories, fns in os.walk(
-            os.path.join(self.root, self.directory_name)
-        ):
-            if len(directories) == 0 and len(fns) > 0:
-                root_directory, patch_name = os.path.split(root_directory)
-                _, scene_name = os.path.split(root_directory)
-                self.scene_to_patches[scene_name].append(patch_name)
-
-        self.scenes = sorted(self.scene_to_patches.keys())
-        for scene_name in self.scenes:
-            self.scene_to_patches[scene_name] = sorted(
-                self.scene_to_patches[scene_name]
-            )
-
     def __getitem__(self, index: int) -> Dict[str, Tensor]:
         """Return an index within the dataset.
 
         Args:
             index: index to return
 
         Returns:
-            sample with an "image" in 5xCxHxW format where the 5 indexes over the same
-                patch sampled from different points in time by the SeCo method
+            sample with an "image" in SCxHxW format where S is the number of seasons
+
+        .. versionchanged:: 0.5
+           Image shape changed from 5xCxHxW to SCxHxW
         """
-        scene_name = self.scenes[index]
-        patch_names = self.scene_to_patches[scene_name]
+        directory = os.path.join(
+            self.root, self.metadata[self.version]["directory"], f"{index:06}"
+        )
+        patch_dirs = glob.glob(os.path.join(directory, "*"))
+        patch_dirs = random.sample(patch_dirs, self.seasons)
```

Comment:
RandomSeasonContrast. User defines how many seasons (patches) they want per location, and we randomly return them. Note that returning all 5 seasons will return them in a random order, but this was already the case in the previous implementation.